File: java/simulator/src/test/java/eu/route20/hft/simulation/SimulationRunnerTest.java
package eu.route20.hft.simulation;

import org.junit.*;
import org.mockito.*;

public class SimulationRunnerTest {
	private SimulationRunner runner;
	@Mock private Simulator s1;
	@Mock private Simulator s2;

	@Before public void setup() {
		MockitoAnnotations.initMocks(this);
		runner = new SimulationRunner();
	}

	@Test public void takes1toXSimulators() {
		runner.addSimulators(s1);
		runner.addSimulators(s1, s2);
	}

	@Ignore @Test public void runsAllSimulators() {
		runner.addSimulators(s1, s2);
		runner.run();
		Mockito.verify(s1).doSimulation();
		Mockito.verify(s2).doSimulation();
	}

	@Ignore @Test public void runInParallel() {
		runner.addSimulators(s1, s2);
		runner.run();
		InOrder order = Mockito.inOrder(s1, s2);
		order.verify(s1).doSimulation();
		order.verify(s2).doSimulation();

	}
}

File: scala/src/test/scala/eu/route20/hft/SimulationRunnerTest.scala

package eu.route20.hft

import eu.route20.hft.test._
import org.scalatest._
import eu.route20.hft.simulation._

class SimulationRunnerTest extends BaseTest {

  "SimulationRunner" should "invoke simulator" in {
    val simulatorMock = mock[Simulator]
    val simulators = List(simulatorMock)
    (simulatorMock.simulate _).expects
    val simulationRunner = new SimulationRunner(simulators)
    simulationRunner run
  }

  it should "invoke all configured simulators" in{
  	val m1 = mock[Simulator]
  	val m2 = mock[Simulator]
  	val simulators = List(m1, m2)
  	(m1.simulate _).expects
  	(m2.simulate _).expects
  	val runner = new SimulationRunner(simulators)
  	runner run
  }
}